### PR TITLE
feat: create lose exp based on rules from tibia

### PR DESCRIPTION
**Pull request by alissonflyncas**
## How?

- Created lose exp on death

## Refs
- https://tibia.fandom.com/wiki/Death#Skill_and_Experience_Loss
- https://www.manjatibia.com.br/calculadora-de-perdas-em-morte/
- https://guildstats.eu/character-level?lang=en

### CODE REVIEW

File: src/Game/NeoServer.Game.Creatures/Model/Players/Player.cs [hunk 1]
```diff
@@ -763,6 +764,21 @@ public bool CanMoveThing(Location location)
         return Location.GetSqmDistance(location) <= MapConstants.MAX_DISTANCE_MOVE_THING;
     }
 
+    public void OnLevelRegress(SkillType type, int fromLevel, int toLevel)
```
Comment (caioavidal) [inline src/Game/NeoServer.Game.Creatures/Model/Players/Player.cs]: shouldn't this method be private?

File: src/Game/NeoServer.Game.Creatures/Model/Players/Player.cs [hunk 2]
```diff
@@ -932,11 +954,24 @@ public bool FlagIsEnabled(PlayerFlag flag)
         return (_flags & (ulong)flag) != 0;
     }
 
+    public void DecreaseExp()
+    {
+        var lostExperience = CalcDecreaseExp();
```
Comment (caioavidal) [inline src/Game/NeoServer.Game.Creatures/Model/Players/Player.cs]: can we rename here to CalculateLostExperience() ?

File: src/NeoServer.Server.Standalone/appsettings.Local.json [hunk 3]
```diff
@@ -3,6 +3,6 @@
     "data": "../../data"
   },
   "database": {
-    "active": "INMEMORY"
+    "active": "SQLITE"
```
Comment (caioavidal) [inline src/NeoServer.Server.Standalone/appsettings.Local.json]: can we remove this file from PR?

File: src/Game/NeoServer.Game.Creatures/Model/Bases/WalkableCreature.cs [hunk 4]
```diff
@@ -174,12 +174,12 @@ public byte[] GetRaw(IPlayer playerRequesting)
 
     public void IncreaseSpeed(ushort speed)
     {
-        ChangeSpeed(speed + Speed);
+        ChangelevelSpeed(speed + Speed);
```
Comment (caioavidal) [inline src/Game/NeoServer.Game.Creatures/Model/Bases/WalkableCreature.cs:177]: can you rename to ChangeSpeedLevel ? but why renaming it btw?

File: src/Server/NeoServer.Server.Events/Player/PlayerLevelRegressedEventHandler.cs [hunk 5]
```diff
@@ -9,27 +9,39 @@
...
             TextMessageOutgoingType.MESSAGE_EVENT_LEVEL_CHANGE));
-    }
-
-    public override void SendLevelChangeMessage(SkillType skillType, IConnection connection, int toLevel)
-    {
-        if (skillType == SkillType.Level) return;
 
         connection.OutgoingPackets.Enqueue(new TextMessagePacket(
-            MessageParser.GetSkillAdvancedMessage(skillType, toLevel),
+            MessageParser.GetSkillRegressedMessage(skillType, toLevel),
```
Comment (caioavidal) [inline src/Server/NeoServer.Server.Events/Player/PlayerLevelRegressedEventHandler.cs]: I think the message should be sent only when you have any difference between fromLevel and toLevel.
If they are the same there is no need for showing the message I think, right?

**Review by caioavidal — CHANGES_REQUESTED** [review]

File: src/Server/NeoServer.Server.Events/Player/PlayerLevelRegressedEventHandler.cs (hunk 5 above)
Comment (alissonflyncas) [reply src/Server/NeoServer.Server.Events/Player/PlayerLevelRegressedEventHandler.cs]: Yes, but whoever calls this class is validating this

**Comment by caioavidal** [comment]
already implemented by #525

**Pull request closed without merging**